tests/mock_network/mod.rs

```diff
@@ -245,7 +245,12 @@ fn node_joins_in_front() {
     verify_invariant_for_all_nodes(&network, &mut nodes);
 }
 
+// Only run for mock parsec, as with DKG Joining node times waiting for NodeApproval.
+// Elder go on to process it and take it on as an Elder.
+// This would not be an issue if Joining did not time out, or if elder processed them quicker.
```

Comment:
There is a way to disable the join timeouts using `poll_and_resend_with_options(nodes, PollOptions::default().fire_join_timeout(false))`. Wouldn't that help?

Reply:
Not really, what fire_join_timeout do is move the clock forward if nothing more is left to do.
What happens here is, with gossip every 1 second, Parsec takes to many seconds to reach the different stage of consensus (i.e it has to add 5 nodes consensusing DKG + SectionInfo, and that takes too many round of gossip then the join node time out).

Reply:
Sorry I lost the reply in github.
fire_join_timeout move the fake_clock forward when nothing is left to do.
This is not what happens here, we gossip every 1 second, and parsec takes to many round of gossip to reach consensus on the 5 nodes we are adding for DKG + SectionInfo, while parsec still gossip we pass the Join timeout.

Option includes increasing Join timeout OR decreasing Parsec gossip time OR avoid doing so many DKG/SectionInfo before adding the last adult.

Reply:
I see. Isn't this going to be a problem in the real network then? Should we make the join timeout longer?

Reply:
I think the solution will come from promotion/demotion and "avoid doing so many DKG/SectionInfo before adding the last adult." i.e add all adults in the queue first before promoting any elder.

Reply:
Ninja'd :)